_posts/2015-12-03-journey_to_high_availability.md

```diff
@@ -0,0 +1,122 @@
+
+- Integrate with DHCP and DNS
+  - Clustered MS DHCP 
+    - One cluster in our main data center
+  - MS DNS
+    - DNS servers per location
+
+Part of the goal is to have Foreman manage our servers lifecycle and increase our efficiency and consistency in provisioning. We can provision without IP addresses and nobody is going to remember the IP addresses of all of their servers. An added benefit is that Foreman cleans these up when hosts are destroyed; something humans rarely do. . . or at least remember to do.
+
+###The Implementation
+
+####Building the Infrastructure
+
+I will discuss building the entire deployment with the exception of the database. Our Enterprise Data Management team built and maintains our PostgreSQL backend; we just have a username, database name and password to connect with. They are using PostgreSQL 9.x and utilizing PG_Pool for high availability.
+
+####Foreman and Memcache (The fun stuff)
+
+The first thing we needed to do before building out our Foreman cluster was to actually figure out what was needed and how Foreman works on a deeper level. Naturally the first move was to go directly to the community in IRC and the Google groups to consult with others that know more than I do. There are two ways you can build your Foreman cluster; a single cluster for all functions or two separate clusters broken out for ENC/Reporting and a webUI. After doing some research into each we decided to go with the simpler single cluster knowing that in the future (sooner than we are expecting) we will need to build a second cluster dedicated to ENC/Reporting functions. Luckily this is an easy change in architecture to implement and can be done with little impact. That is for another post though, so I will focus on what we ended up doing. After some consideration we decided to go with a three node cluster (by the time this gets published it might very well be a four node cluster). We were going to originally go with a two node cluster but since our legacy stuff makes heavy use of smart-class parameters we wanted the extra node to help spread the load rendering all of the ENC data for hosts we aren't going to migrate to our new code base. We also learned that Foreman stores it's sessions in the database so loadbalancing wasn't going to be as complex as we originally thought and we could actually look at implementing some cool things in the future since we don't need to implement any type of persistence. Another important thing to know is that each Foreman node will have their hostname based Puppet certificate, but for clustering you will need to generate a generically named certificate to be distributed across your nodes and used for the webUI and ENC/Reporting functions. We generated the certificate based on the URL name we were going to use. 
+
```

Comment:
Right, thats for outbound traffic to the Proxies etc, yes? I just want to avoid confusion, as its written I could interpret that as *every* host in Foreman, which I don't think is right :)